active-rfcs/0000-finalize-head-body.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 11/30/2021
+- Reference Issues: https://github.com/withastro/rfcs/discussions/15
+- Implementation PR:
+
+# Summary
+
+Finalize behavior around `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` elements in Astro component templates.
+
+# Motivation
+
+in Astro v0.21, we have had user reports around `<head>` not acting as expected, breaking some projects from v0.20:
+- https://github.com/withastro/astro/issues/2046
+- https://github.com/withastro/astro/issues/2022
+
+Some of these issues stem from undocumented or undefined behaviors in v0.20, like the fact that `<head>` could be conditional.
+
+This RFC is an attempt to agree on and document the behavior of the `<head>`, `<body>`, and `<html>` tags in a way that makes sense, and can be communicated to users. It should support most common use-cases, such as layouts:
+
+```astro
+---
+// src/pages/index.astro
+import CommonLayout from '../layouts/CommonLayout.astro';
+---
+<!-- this is valid, where CommonLayout contains `<head>`, `<body>`, etc.
+<CommonLayout someProp="..." />
+```
+
+## Goals
+
+- Agree on and document the rules and behaviors of the `<head>`, `<body>`, and `<html>` tags.
+- Support any component in the tree adding to the page `<head>`, similar to `<svelte:head>` or head manager libraries in React/Vue. 
+  - (Note: see "Alternatives" for an alternative design if this is decided to be a non-goal).
+
+## Non-Goals
```

Comment:
@jonathantneal reworded this section to make it more clear that this proposal doesn't support optional `<head>` and `<body>` tags in the output, and why. 

Since some of your comments were rewording parts of the proposal to add support for this, it might be better to discuss any non-goals that you disagree with here first, since those changes didn't make sense without first changing the goals/non-goals here.